src/System.Net.Http/src/System/Net/Http/HttpUtilities.cs

```diff
@@ -25,12 +25,21 @@ internal static class HttpUtilities
         internal static bool IsHttpUri(Uri uri)
         {
             Debug.Assert(uri != null);
-
-            string scheme = uri.Scheme;
-            return string.Equals("http", scheme, StringComparison.OrdinalIgnoreCase) ||
-                string.Equals("https", scheme, StringComparison.OrdinalIgnoreCase);
+            return IsSupportedScheme(uri.Scheme);
         }
 
+        internal static bool IsSupportedScheme(string scheme) =>
+            IsSupportedNonSecureScheme(scheme) ||
+            IsSupportedSecureScheme(scheme);
+
+        internal static bool IsSupportedNonSecureScheme(string scheme) =>
+            string.Equals(scheme, "http", StringComparison.OrdinalIgnoreCase) ||
+            string.Equals(scheme, "ws", StringComparison.OrdinalIgnoreCase);
+
+        internal static bool IsSupportedSecureScheme(string scheme) =>
+            string.Equals(scheme, "https", StringComparison.OrdinalIgnoreCase) ||
+            string.Equals(scheme, "wss", StringComparison.OrdinalIgnoreCase);
```

Comment:
What if the websocket schemes are used directly from HttpClient?

Reply:
They'd be allowed.

Reply:
Is this what we want?  e.g. is this what the WinHttpHandler does?

I would have expected the ClientWebSocket logic to construct a Uri with an http scheme (http/https as appropriate) and then pass this down to HttpClient.

Reply:
> I would have expected the ClientWebSocket logic to construct a Uri with an http scheme (http/https as appropriate) and then pass this down to HttpClient.

How does that work when going through a proxy?

Reply:
> How does that work when going through a proxy? Wouldn't the proxy then forward the wrong scheme as part of the request?

That said, if it works fine with proxies, changing the scheme was my original plan and is certainly easier.